### PR TITLE
8239785: Cgroups: Incorrect detection logic on old systems in hotspot

### PR DESCRIPTION
**Pull request by jmtd**
This is a backport of 8239785 for jdk11u. The dependent backports have now all been integrated.

This PR replaces #881 which was stuck with a dependency on a superseded dependent PR.

It's not clean, but the only conflict was the copyright line in the
comment header for src/hotspot/share/prims/whitebox.hpp.

I'm checking jtreg tests in test/hotspot/jtreg/containers on a mixture of cgroup v1 and v2 systems now.

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [ ] Change must be properly reviewed

### Error
&nbsp;⚠️ The pull request body must not be empty.

### Issue
 * [JDK-8239785](https://bugs.openjdk.java.net/browse/JDK-8239785): Cgroups: Incorrect detection logic on old systems in hotspot

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/jdk11u-dev pull/946/head:pull/946` \
`$ git checkout pull/946`

Update a local copy of the PR: \
`$ git checkout pull/946` \
`$ git pull https://git.openjdk.java.net/jdk11u-dev pull/946/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 946`

View PR using the GUI difftool: \
`$ git pr show -t 946`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/jdk11u-dev/pull/946.diff">https://git.openjdk.java.net/jdk11u-dev/pull/946.diff</a>

</details>

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8239785](https://bugs.openjdk.java.net/browse/JDK-8239785): Cgroups: Incorrect detection logic on old systems in hotspot

### Reviewers
 * [Severin Gehwolf](https://openjdk.java.net/census#sgehwolf) (@jerboaa - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/jdk11u-dev pull/946/head:pull/946` \
`$ git checkout pull/946`

Update a local copy of the PR: \
`$ git checkout pull/946` \
`$ git pull https://git.openjdk.java.net/jdk11u-dev pull/946/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 946`

View PR using the GUI difftool: \
`$ git pr show -t 946`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/jdk11u-dev/pull/946.diff">https://git.openjdk.java.net/jdk11u-dev/pull/946.diff</a>

</details>

### CODE REVIEW

File: test/lib/sun/hotspot/WhiteBox.java [hunk 1]
```diff
@@ -1,5 +1,5 @@
 /*
- * Copyright (c) 2012, 2019, Oracle and/or its affiliates. All rights reserved.
+ * Copyright (c) 2012, 2020, Oracle and/or its affiliates. All rights reserved.
```
Comment (jerboaa) [inline test/lib/sun/hotspot/WhiteBox.java:2]: This change isn't in the original patch. Please remove.

**Review by jerboaa — CHANGES_REQUESTED** [review]
[src/hotspot/share/prims/whitebox.hpp](https://github.com/openjdk/jdk11u-dev/pull/946/files#diff-0616ebadb392e16dc81358b7c8e2cdf30e171b91402bedee9aeeb0a4f57c25cc) Doesn't have the Copyright update unlike the original patch. Please add. `WhiteBox.java` on the other hand didn't have the copyright bumped in the original patch, so neither should we here.

Other than that. Looks fine.

File: test/lib/sun/hotspot/WhiteBox.java (hunk 1 above)
Comment (jmtd) [reply test/lib/sun/hotspot/WhiteBox.java:2]: Odd sorry, I picked the wrong whitebox file!

**Comment by jmtd** [comment]
Testing notes:

On a cgroups v2 system with glibc 2.33, specifying `-Djdk.test.docker.image.version=hirsute` (the version of ubuntu with a matching libc version),  8 tests pass; 1 fails - containers/cgroup/PlainRead.java - which is expected (to be fixed in backport of [JDK-8278951](https://bugs.openjdk.java.net/browse/JDK-8278951))

Running tests on a cgroups v1 system now.

**Comment by jerboaa** [comment]
I'm marking this one clean as the only change is the copyright line in `whitebox.hpp` which has been fixed manually.

**Review by jerboaa — APPROVED** [review]

**Comment by jerboaa** [comment]
@jmtd Please request for backport approval on the bug.

**Pull request closed without merging**